packages/web/components/table/cells/pool-composition.tsx

```diff
@@ -1,54 +1,64 @@
+import classNames from "classnames";
 import Image from "next/image";
 import React, { FunctionComponent } from "react";
-import classNames from "classnames";
 import { BaseCell } from "..";
+import { useStore } from "../../../stores";
+import { PoolAssetsIcon } from "../../assets";
 
-export interface PoolCompositionCell {
-  poolId: number;
-  tokenDenoms: string[];
+export interface PoolCompositionCell extends BaseCell {
+  poolId: string;
 }
 
 /** Displays pool composition as a cell in a table.
  *
  *  Accepts the base hover flag.
  */
-export const PoolCompositionCell: FunctionComponent<
-  Partial<BaseCell & PoolCompositionCell>
-> = ({ value, rowHovered, poolId, tokenDenoms }) => (
-  <React.Fragment>
-    {poolId && tokenDenoms ? (
-      <div className="flex gap-1">
-        <div className="flex flex-col items-start text-white-full">
-          <span
-            className={classNames({
-              "text-secondary-200": rowHovered,
-            })}
-          >
-            {tokenDenoms.reduce(
-              (str, denom, index) =>
-                index === tokenDenoms.length - 1
-                  ? str + denom
-                  : str + denom + "/",
-              ""
-            )}
-          </span>
-          <span
-            className={classNames("text-sm font-caption opacity-60", {
-              "text-secondary-600": rowHovered,
-            })}
-          >
-            Pool #{poolId}
-          </span>
+export const PoolCompositionCell: FunctionComponent<PoolCompositionCell> = ({
+  value,
+  rowHovered,
+  poolId,
+}) => {
+  const { queriesOsmosisStore, chainStore } = useStore();
+
+  const queryOsmosis = queriesOsmosisStore.get(chainStore.osmosis.chainId);
+
+  const pool = queryOsmosis.queryGammPools.getPool(poolId);
```

Comment:
My thought is it would help decouple the view from where the data comes from, even this is performant. If we refactor the structure of the queries it may be easier to continue to use this component.

TLDR: remove a dependency here.